### PR TITLE
Feat: Awakening Packet, SetJob Command

### PR DESCRIPTION
**Pull request by Clockworkx**
### SetJob Command
Syntax: /setJob class awakening=1 or unawakened =0
E.g. /setJob Assassin 1

### Misc
Some improvement on commandInfo / commands

### CODE REVIEW

File: MapleServer2/Types/Player.cs [hunk 1]
```diff
@@ -28,9 +28,8 @@ public class Player
         public byte Gender { get; set; }
 
         // Job Group, according to jobgroupname.xml
-        public bool Awakened { get; set; }
         public Job Job { get; set; }
-        public JobCode JobCode => (JobCode) ((int) Job * 10 + (Awakened ? 1 : 0));
+        public JobCode JobCode { get; set; }
```
Comment (Zintixx) [inline MapleServer2/Types/Player.cs:32]: hell yea

File: MapleServer2/Packets/JobPacket.cs [hunk 2]
```diff
@@ -89,5 +89,61 @@ public static Packet WritePassiveSkills(PacketWriter pWriter, IFieldObject<Playe
...
+        //        pWriter.WriteByte();
+        //        pWriter.WriteBool(skills[id] > 0);  // Is it learned?
+        //        pWriter.WriteInt(id);               // Skill to display
+        //        pWriter.WriteInt(Math.Clamp(skills[id], skillData[id].SkillLevels.Select(x => x.Level).FirstOrDefault(), int.MaxValue));    // Level to display
+        //        pWriter.WriteByte();
+        //    }
+        //    pWriter.WriteShort(); // Ends with zero short
+
+        //    return pWriter;
+        //}
```
Comment (Zintixx) [inline MapleServer2/Packets/JobPacket.cs:147]: Can we remove this for master?

**Review by Zintixx — CHANGES_REQUESTED** [review]

File: MapleServer2/Packets/JobPacket.cs (hunk 2 above)
Comment (Clockworkx) [reply MapleServer2/Packets/JobPacket.cs:147]: omg im a dummy sorry haha

File: MapleServer2/Commands/Game/GMCommands.cs [hunk 3]
```diff
@@ -26,4 +31,64 @@ public override void Execute(GameCommandTrigger trigger)
...
+
+            Player player = trigger.Session.Player;
+            IFieldObject<Player> fieldPlayer = trigger.Session.FieldPlayer;
+
+            long activeSkillTabId = player.ActiveSkillTabId;
+            SkillTab skillTab = player.SkillTabs.First(tab => tab.TabId == activeSkillTabId);
+
+            Job job = Job.None;
+
+
```
Comment (AlanMorel) [inline MapleServer2/Commands/Game/GMCommands.cs:62]: blank line

File: MapleServer2/Commands/Game/InfoCommands.cs [hunk 4]
```diff
@@ -29,13 +29,22 @@ public override void Execute(GameCommandTrigger trigger)
...
                 {
-                    stringBuilder.Append(CommandHelpers.Color(CommandHelpers.Bold(alias), Color.DarkOrange));
+                    stringBuilder.Append(CommandHelpers.Color(CommandHelpers.Bold(alias), Color.DarkOrange) + " ");
                 }
 
                 stringBuilder.Append($"{CommandHelpers.Color(command.Description, Color.Wheat)}\n");
+                //if (command != commandList.Last())
+                //{
+                //    stringBuilder.Append('\n');
+                //}
```
Comment (AlanMorel) [inline MapleServer2/Commands/Game/InfoCommands.cs:39]: Can this be removed if its just commented out?

File: MapleServer2/Commands/Game/GMCommands.cs [hunk 5]
```diff
@@ -26,4 +31,64 @@ public override void Execute(GameCommandTrigger trigger)
...
+            if (!Enum.TryParse(jobName, true, out job))
```
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/GMCommands.cs:63]:
```suggestion
            Job job = Job.None;

            if (!Enum.TryParse(jobName, true, out job))
```

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Commands/Game/GMCommands.cs [hunk 6]
```diff
@@ -26,4 +31,64 @@ public override void Execute(GameCommandTrigger trigger)
...
+                "sj"
+            };
+            Description = "Sets character's job, e.g. /setJob assassin 1";
+            AddParameter<string>("job", "Classname, e.g.: striker");
+            AddParameter<byte>("awakened", "Awakened = 1, Unawakened = 0");
+        }
+
+        public override void Execute(GameCommandTrigger trigger)
+        {
+
```
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/GMCommands.cs:50]: blank line

File: MapleServer2/Commands/Game/GMCommands.cs [hunk 7]
```diff
@@ -26,4 +31,64 @@ public override void Execute(GameCommandTrigger trigger)
...
+                player.Session.Send(NoticePacket.Notice($"{CommandHelpers.Color(CommandHelpers.Bold("You have to give a classname and specifiy awakening (1 or 0)\nAvailable classes:\n"), Color.DarkOrange)}" +
+                    $"{CommandHelpers.Color(string.Join(", ", classes), Color.Aquamarine)}", NoticeType.Chat));
+
+                return;
+            }
+
+            JobCode jobCode = (JobCode) ((int) job * 10 + awakened);
+            player.JobCode = jobCode;
+
+            if (job != Job.None && job != player.Job)
```
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/GMCommands.cs:82]: not needed
```suggestion
            if (job != player.Job)
```

File: MapleServer2/Commands/Game/GMCommands.cs [hunk 8]
```diff
@@ -26,4 +31,64 @@ public override void Execute(GameCommandTrigger trigger)
...
+            {
+                DatabaseManager.SkillTabs.Delete(skillTab.Uid);
+                SkillTab newSkillTab = new SkillTab(player.CharacterId, job, skillTab.TabId, skillTab.Name);
+
+                player.SkillTabs[player.SkillTabs.IndexOf(skillTab)] = newSkillTab;
+                player.Job = job;
+            }
+            trigger.Session.Send(Packets.JobPacket.SendJob(fieldPlayer));
+        }
+
```
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/GMCommands.cs:92]: blank line

File: MapleServer2/Commands/Game/GMCommands.cs [hunk 9]
```diff
@@ -26,4 +31,64 @@ public override void Execute(GameCommandTrigger trigger)
...
+            Job job = Job.None;
+
+
+            if (!Enum.TryParse(jobName, true, out job))
+            {
+                player.Session.SendNotice($"{jobName} is not a valid class name");
+                return;
+            }
+
+            if (Job.None == job)
```
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/GMCommands.cs:69]: This if will ever get reached since the if above returns if it can't parse? Only if you use `/sj None` correct?

File: MapleServer2/Commands/Game/InfoCommands.cs (hunk 4 above)
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/InfoCommands.cs:39]: remove or add reason why it's commented out

File: MapleServer2/Commands/Game/InfoCommands.cs [hunk 10]
```diff
@@ -29,13 +29,22 @@ public override void Execute(GameCommandTrigger trigger)
...
                 foreach (IParameter param in command.Parameters)
                 {
-                    stringBuilder.Append($" - {CommandHelpers.Color(param.Name, Color.Aquamarine)} {CommandHelpers.Color(param.Description, Color.Wheat)}\n");
+
+                    stringBuilder.Append($"    -{CommandHelpers.Color(param.Name, Color.Aquamarine)} {CommandHelpers.Color(param.Description, Color.Wheat)}");
```
Comment (AngeloTadeucci) [inline MapleServer2/Commands/Game/InfoCommands.cs:43]:
```suggestion
                    stringBuilder.Append($"    -{CommandHelpers.Color(param.Name, Color.Aquamarine)} {CommandHelpers.Color(param.Description, Color.Wheat)}");
```

File: MapleServer2/Commands/Game/InfoCommands.cs (hunk 4 above)
Comment (Clockworkx) [reply MapleServer2/Commands/Game/InfoCommands.cs:39]: sorry ... I didn't see it 🙈

File: MapleServer2/Constants/Skills/SkillTreeOrdered.cs [hunk 11]
```diff
@@ -67,10 +67,10 @@ public static class SkillTreeOrdered
         };
         private static readonly List<int> Assassin = new List<int>
         {
-            10800101,10800271,10800203,10800051,10800221,10800001,10800171,20000011,10800121,10800291,
+            10800251,10800271,10800203,10800051,10800221,10800001,10800171,20000011,10800121,10800291,
             10800071,10800021,10800191,10800293,10800141,10800261,10800041,10800211,10800161,10800229,
             20000001,10800281,10800061,10800231,10800011,10800181,10800131,10800081,10800031,10800201,
-            10800151,10800052,10800241,10800091,10800111,10800163,10800232,10800251
+            10800151,10800052,10800241,10800091,10800111,10800163,10800232,10800101
```
Comment (AngeloTadeucci) [inline MapleServer2/Constants/Skills/SkillTreeOrdered.cs:73]: revert this changes

File: MapleServer2/Constants/Skills/SkillTreeOrdered.cs [hunk 12]
```diff
@@ -107,8 +107,20 @@ public static class SkillTreeOrdered
...
         };
+        private static readonly List<int> Beginner = new List<int>
+        {
+            10000001,
+            10000011,
+            10000021,
+            10000031,
+            10000041,
+            10000051,
+            10000061,
```
Comment (AngeloTadeucci) [inline MapleServer2/Constants/Skills/SkillTreeOrdered.cs:121]:
```suggestion
            10000001, 10000011, 10000021, 10000031, 10000041, 10000051, 10000061
```

File: MapleServer2/Commands/Game/GMCommands.cs (hunk 3 above)
Comment (Clockworkx) [reply MapleServer2/Commands/Game/GMCommands.cs:62]: removed!

File: MapleServer2/Data/Static/SkillMetadataStorage.cs [hunk 13]
```diff
@@ -27,7 +28,19 @@ public static SkillMetadata GetSkill(int id)
         // Get a List of Skills corresponding to the Job
         public static List<SkillMetadata> GetJobSkills(Job job = Job.None)
         {
+
```
Comment (AngeloTadeucci) [inline MapleServer2/Data/Static/SkillMetadataStorage.cs:31]:
```suggestion
```

File: MapleServer2/Packets/FieldPacket.cs [hunk 14]
```diff
@@ -39,7 +39,7 @@ public static Packet AddPlayer(IFieldObject<Player> fieldPlayer)
             CharacterListPacket.WriteCharacter(player, pWriter);
 
             // Skills
-            pWriter.WriteEnum(player.JobCode);
+            pWriter.WriteEnum(fieldPlayer.Value.JobCode);
```
Comment (AngeloTadeucci) [inline MapleServer2/Packets/FieldPacket.cs:42]: This change is really needed?

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/Player.cs (hunk 1 above)
Comment (Clockworkx) [reply MapleServer2/Types/Player.cs:32]: zin :( they didn't like it :sadge:
we back to formula :D

**Review by AngeloTadeucci — APPROVED** [review]
LGTM

**Review by Sparkymod — APPROVED** [review]

**Review by AngeloTadeucci — APPROVED** [review]
